source/partialObject.js

```diff
@@ -0,0 +1,32 @@
+import mergeDeepRight from './mergeDeepRight';
+import _curry2 from './internal/_curry2';
+
+/**
+ * Takes a function `f` and an object, and returns a function `g`.
+ * When applied, `g` returns the result of applying `f` to the object
+ * provided initially merged deeply (right) with the object provided as an argument to `g`.
+ *
+ * @func
+ * @memberOf R
+ * @category Function
+ * @sig (({ a, b, c, ..., n }) -> x) -> { a, b, c, ...} -> ({ d, e, f, ..., n } -> x)
+ * @param {Function} f
+ * @param {Object} props
+ * @return {Function}
+ * @see R.partial, R.partialRight, R.curry, R.mergeDeepRight
+ * @example
+ *
+ *      const multiply2 = ({ a, b }) => a * b;
+ *      const double = R.partialObject(multiply2, { a: 2 });
+ *      double({ b: 2 }); //=> 4
+ *
+ *      const greet = ({ salutation, title, firstName, lastName }) =>
+ *        salutation + ', ' + title + ' ' + firstName + ' ' + lastName + '!';
+ *
+ *      const sayHello = R.partialObject(greet, { salutation: 'Hello' });
+ *      const sayHelloToMs = R.partialObject(sayHello, { title: 'Ms.' });
+ *      sayHelloToMs({ firstName: 'Jane', lastName: 'Jones' }); //=> 'Hello, Ms. Jane Jones!'
+ * @symb R.partial(f, { a, b })({ c, d }) = f({ a, b, c, d })
+ */
+
+export default _curry2((f, o) => (props) => f.call(this, mergeDeepRight(o, props)));
```

Comment:
Do you really need this function? Note that `partialObject(f)` is equivalent to `curry(compose(f, mergeDeepRight))`.

Here's a [demo](https://ramdajs.com/repl/?v=0.27.0#?const%20discriminant%20%3D%20%28%7B%20a%2C%20b%2C%20c%20%7D%29%20%3D%3E%20b%20%2A%20b%20-%204%20%2A%20a%20%2A%20c%3B%0Aconst%20f%20%3D%20curry%28compose%28discriminant%2C%20mergeDeepRight%29%29%3B%0A%0A%2F%2F%20it%20caches%20the%20initially%20supplied%20arguments%0A%0Aconsole.log%28f%28%7B%20a%3A%203%20%7D%29%28%7B%20b%3A%207%2C%20c%3A%204%20%7D%29%29%3B%20%2F%2F%201%0Aconsole.log%28f%28%7B%20a%3A%203%2C%20b%3A%207%20%7D%29%28%7B%20c%3A%204%20%7D%29%29%3B%20%2F%2F%201%0A%0A%2F%2F%20it%20works%20regardless%20of%20property%20order%0A%0Aconsole.log%28f%28%7B%20b%3A%207%20%7D%29%28%7B%20c%3A%204%2C%20a%3A%203%20%7D%29%29%3B%20%2F%2F%201%0Aconst%20g%20%3D%20curry%28compose%28f%28%7B%20a%3A%203%20%7D%29%2C%20mergeDeepRight%28%7B%20c%3A%204%20%7D%29%29%29%3B%0Aconsole.log%28g%28%7B%20b%3A%207%20%7D%29%29%3B%20%2F%2F%201%0A%0A%2F%2F%20it%20merges%20deeply%20right%0A%0Aconst%20nestedDiscriminant%20%3D%20%28%7B%20a%2C%20c%2C%20d%20%7D%29%20%3D%3E%20d.b%20%2A%20d.b%20-%204%20%2A%20a%20%2A%20c%3B%0Aconst%20h%20%3D%20curry%28compose%28nestedDiscriminant%2C%20mergeDeepRight%29%29%3B%0Aconst%20j%20%3D%20curry%28compose%28h%2C%20mergeDeepRight%28%7B%20a%3A%200%2C%20d%3A%20%7B%20b%3A%200%20%7D%20%7D%29%29%29%3B%0Aconsole.log%28j%28%7B%20a%3A%203%2C%20d%3A%20%7B%20b%3A%207%20%7D%20%7D%29%28%7B%20c%3A%204%20%7D%29%29%3B%20%2F%2F%201%0A). I think `curry(compose(...))` offers more flexibility because you can choose which merging function to use.

```javascript
const discriminant = ({ a, b, c }) => b * b - 4 * a * c;
const f = curry(compose(discriminant, mergeDeepRight));

// it caches the initially supplied arguments

console.log(f({ a: 3 })({ b: 7, c: 4 })); // 1
console.log(f({ a: 3, b: 7 })({ c: 4 })); // 1

// it works regardless of property order

console.log(f({ b: 7 })({ c: 4, a: 3 })); // 1
const g = curry(compose(f({ a: 3 }), mergeDeepRight({ c: 4 })));
console.log(g({ b: 7 })); // 1

// it merges deeply right

const nestedDiscriminant = ({ a, c, d }) => d.b * d.b - 4 * a * c;
const h = curry(compose(nestedDiscriminant, mergeDeepRight));
const j = curry(compose(h, mergeDeepRight({ a: 0, d: { b: 0 } })));
console.log(j({ a: 3, d: { b: 7 } })({ c: 4 })); // 1
```

The point is that you really don't need a `partialObject` or even a `namedCurry` function.

Reply:
That's a valid point, thanks for the examples.

I suppose the question, in the words of @CrossEye , is whether the utility of this is enough to make a library implementation useful?

I personally think the utility of having this function is worthwhile, and doing it manually with `curry`, `compose` and a merge function is a great option when I need something more specific. But of course I'm curious to hear other thoughts as well.

Reply:
@aaditmshah:

Yes, it's clear that this function is not necessary.  But that's the nature of a library: anything it does can be done directly in your own code.

Ramda strives to strike a balance between several different factors, chiefly how widely useful a function is and how difficult or easy it would be for users if we kept it in userland.

One of my favorite examples is that we have both `pipe` and `compose`.  We could obviously keep only one of them as either is trivial to write in terms of the other.  But they are both so useful and they are often used by different audiences, so we supply both of them.  We can contrast that with [diffobjs][do], a much less trivial-to-write function but which seems to have a much more restricted audience.  In that case, we don't supply it, although we keep a link in the [Cookbook][cb].

I was mixed about whether we should include this functionality in the earlier guises: one which specified what properties/paths needed to be included before applying the underlying function.  But with this formulation, I see it immediately as something I've written many times in one way or another.  It strikes me as one likely to get a fair bit of use.



  [cb]: https://github.com/ramda/ramda/wiki/Cookbook
  [do]: https://github.com/ramda/ramda/wiki/Cookbook#diffobjs---diffing-objects-similar-to-guavas-mapsdifference